org.lflang/src/org/lflang/generator/c/CTriggerObjectsGenerator.java

```diff
@@ -343,30 +343,42 @@ private static boolean setReactionPriorities(
     ) {
         var foundOne = false;
         // Force calculation of levels if it has not been done.
+        // FIXME: Why is this repeated all over? Are there scenarios where we haven't assigned levels yet?
```

Comment:
Again assignLevels are called, in case it hasn't been done. It is also invoked each time we call "getLevels"